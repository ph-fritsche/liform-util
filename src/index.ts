export { isShallowEqual } from './comparison/isShallowEqual'
export { intervalFromString, Interval } from './date/interval'
export { getIndicesOfDescendant } from './dom/indicesOfDescendant'
export { useBoundFunction } from './react/boundFunction'
export { useForkedCallback } from './react/forkedCallback'
export { useForkedRef, updateRef } from './react/forkedRef'
export { useId } from './react/id'
export { ComponentForwardRef, WrapperProps, ForwardRefWrapper, createForwardRefWrapper } from './react/wrapper'
export { isEscaped, countUnescaped, hasUnescaped, indexOfUnescaped } from './string/escaped'
export { filterObject } from './object/filterObject'
