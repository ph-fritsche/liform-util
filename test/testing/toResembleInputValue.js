import '../../src/testing'
import { render } from '../_render'

it.each([
    [`<input value=""/>`, [null, undefined, '', false, []], ['0', 0]],
    [`<input value="0"/>`, [0, false, '0'], [null, undefined, []]],
    [`<input value="1"/>`, [1, true, '1'], ['foo']],
    [`<input value="fooValue"/>`, ['fooValue', true], [1]],
    [`<input type="checkbox" value="fooValue" checked/>`, ['fooValue', true], [1, 'true']],
    [`<input type="checkbox" checked/>`, [true, 'true'], ['foo']],
    [`<input type="checkbox" value="fooValue"/>`, ['', null, undefined, false, []], [0]],
    [`<input type="number" value=""/>`, [null, undefined, '', false], [0]],
    [`<input type="number" value="0"/>`, [0, false, '0'], [null, undefined, '']],
    [`<input type="number" value="1"/>`, [1, true, '1'], [1.2]],
    [`<select><option selected>bar</option></select>`, ['bar', ['bar'], true], [[], null, undefined, '', false]],
    [`<select multiple><option selected>bar</option><option selected>baz</option></select>`, [['bar', 'baz'], 'bar,baz', true], [1]],
    [`<select multiple></select>`, [[], false, null, undefined, ''], [0]],
])('Compare input values %p', (html, expectedValues, unexpectedValues) => {
    const { element } = render(html)

    for (const v of expectedValues) {
        expect(element).toResembleInputValue(v)
    }

    for (const v of unexpectedValues) {
        expect(() => {
            expect(element).toResembleInputValue(v)
        }).toThrowErrorMatchingSnapshot()
    }
})
