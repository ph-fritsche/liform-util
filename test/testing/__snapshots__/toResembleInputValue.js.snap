// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare input values "<input type=\\"checkbox\\" checked/>" 1`] = `
"expect(INPUT).toResembleInputValue(\\"foo\\")

Expected: \\"foo\\"
Received: true"
`;

exports[`Compare input values "<input type=\\"checkbox\\" value=\\"fooValue\\" checked/>" 1`] = `
"expect(INPUT).toResembleInputValue(1)

Expected: 1
Received: \\"fooValue\\""
`;

exports[`Compare input values "<input type=\\"checkbox\\" value=\\"fooValue\\" checked/>" 2`] = `
"expect(INPUT).toResembleInputValue(\\"true\\")

Expected: \\"true\\"
Received: \\"fooValue\\""
`;

exports[`Compare input values "<input type=\\"checkbox\\" value=\\"fooValue\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(0)

Expected: 0
Received: undefined"
`;

exports[`Compare input values "<input type=\\"number\\" value=\\"\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(0)

Expected: 0
Received: null"
`;

exports[`Compare input values "<input type=\\"number\\" value=\\"0\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(null)

Expected: null
Received: 0"
`;

exports[`Compare input values "<input type=\\"number\\" value=\\"0\\"/>" 2`] = `
"expect(INPUT).toResembleInputValue(undefined)

Expected: undefined
Received: 0"
`;

exports[`Compare input values "<input type=\\"number\\" value=\\"0\\"/>" 3`] = `
"expect(INPUT).toResembleInputValue(\\"\\")

Expected: \\"\\"
Received: 0"
`;

exports[`Compare input values "<input type=\\"number\\" value=\\"1\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(1.2)

Expected: 1.2
Received: 1"
`;

exports[`Compare input values "<input value=\\"\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(\\"0\\")

Expected: \\"0\\"
Received: \\"\\""
`;

exports[`Compare input values "<input value=\\"\\"/>" 2`] = `
"expect(INPUT).toResembleInputValue(0)

Expected: 0
Received: \\"\\""
`;

exports[`Compare input values "<input value=\\"0\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(null)

Expected: null
Received: \\"0\\""
`;

exports[`Compare input values "<input value=\\"0\\"/>" 2`] = `
"expect(INPUT).toResembleInputValue(undefined)

Expected: undefined
Received: \\"0\\""
`;

exports[`Compare input values "<input value=\\"0\\"/>" 3`] = `
"expect(INPUT).toResembleInputValue([])

Expected: []
Received: \\"0\\""
`;

exports[`Compare input values "<input value=\\"1\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(\\"foo\\")

Expected: \\"foo\\"
Received: \\"1\\""
`;

exports[`Compare input values "<input value=\\"fooValue\\"/>" 1`] = `
"expect(INPUT).toResembleInputValue(1)

Expected: 1
Received: \\"fooValue\\""
`;

exports[`Compare input values "<select multiple></select>" 1`] = `
"expect(SELECT).toResembleInputValue(0)

Expected: 0
Received: []"
`;

exports[`Compare input values "<select multiple><option selected>bar</option><option selected>baz</option></select>" 1`] = `
"expect(SELECT).toResembleInputValue(1)

Expected: 1
Received: [\\"bar\\", \\"baz\\"]"
`;

exports[`Compare input values "<select><option selected>bar</option></select>" 1`] = `
"expect(SELECT).toResembleInputValue([])

Expected: []
Received: \\"bar\\""
`;

exports[`Compare input values "<select><option selected>bar</option></select>" 2`] = `
"expect(SELECT).toResembleInputValue(null)

Expected: null
Received: \\"bar\\""
`;

exports[`Compare input values "<select><option selected>bar</option></select>" 3`] = `
"expect(SELECT).toResembleInputValue(undefined)

Expected: undefined
Received: \\"bar\\""
`;

exports[`Compare input values "<select><option selected>bar</option></select>" 4`] = `
"expect(SELECT).toResembleInputValue(\\"\\")

Expected: \\"\\"
Received: \\"bar\\""
`;

exports[`Compare input values "<select><option selected>bar</option></select>" 5`] = `
"expect(SELECT).toResembleInputValue(false)

Expected: false
Received: \\"bar\\""
`;
